components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.ts

```diff
@@ -91,8 +91,8 @@ export class EnvironmentDetailsComponent implements OnInit, OnDestroy {
     ).subscribe(environment => {
       this.show = true;
       this.environment = environment;
-      this.cookbookVersions = environment.cookbook_versions.filter(Boolean);
-      if (this.cookbookVersions.length > 0) {
+      this.cookbookVersions = environment.cookbook_versions;
```

Comment:
To go along with my previous suggestion, this would change to, for example:
```suggestion
      this.cookbookVersions = this.toDisplay(environment.cookbook_versions);
```
Then add:
```
toDisplay(cookbookVersions: CookbookVersion[]): CookbookVersionDisplay[] {
. . .
}
```

Reply:
updated the changes, please have a look.